string/string.h

```diff
@@ -0,0 +1,297 @@
+/**
+ * Library for embedded strings
+ *
+ * @author Aaron Chan
+ */
+
+#ifndef STRING_H
+#define STRING_H
+
+#include <cstddef>
+
+const char NULL_TERMINATOR = '\0';
+
+template<const size_t size>
+class String {
+public:
+    int strlen = 0;
+    char string[size] = {};
+
+    String() {}
```

Comment:
you should get rid of this constructor, you only want a protected constructor to force all declarations to use an alloc::String